sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1060,16 +1060,33 @@ class GpuRLikeMeta(
     parent: Option[RapidsMeta[_, _, _]],
     rule: DataFromReplacementRule) extends BinaryExprMeta[RLike](expr, conf, parent, rule) {
 
+    private var originalPattern: String = ""
     private var pattern: Option[String] = None
 
+    def optimizeSimplePattern(lhs: Expression, rhs: Expression): GpuExpression = {
+      import RegexOptimizationType._
+      val originalAst = new RegexParser(originalPattern).parse()
+      RegexRewriteUtils.matchSimplePattern(originalAst) match {
+        case StartsWith(s) => GpuStartsWith(lhs, GpuLiteral(s, StringType))
+        case Contains(s) => GpuContains(lhs, GpuLiteral(s, StringType))
+        case NoOptimization => {
+          val patternStr = pattern.getOrElse(throw new IllegalStateException(
+            "Expression has not been tagged with cuDF regex pattern"))
+          GpuRLike(lhs, rhs, patternStr)
+        }
+        case _ => throw new IllegalStateException("Unexpected optimization type")
+      }
+    }
+
     override def tagExprForGpu(): Unit = {
       GpuRegExpUtils.tagForRegExpEnabled(this)
       expr.right match {
         case Literal(str: UTF8String, DataTypes.StringType) if str != null =>
           try {
             // verify that we support this regex and can transpile it to cuDF format
-            val (transpiledAST, _) =
-                new CudfRegexTranspiler(RegexFindMode).getTranspiledAST(str.toString, None, None)
+            originalPattern = str.toString
+            val (transpiledAST, _) = new CudfRegexTranspiler(RegexFindMode)
```

Comment:
nit: Could we have a follow on issue to figure out how to parse the regexp once, instead of multiple times?

Reply:
Filed https://github.com/NVIDIA/spark-rapids/issues/10817, will do it in my next regex rewrite pr.